web/modules/custom/custom_block/src/Controller/ExampleController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\custom_block\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Returns responses for Custom block routes.
+ */
+final class ExampleController extends ControllerBase {
+
+  /**
+   * Builds the response.
+   */
+  public function __invoke(): array {
+
+    $build['content'] = [
+      '#markup' => $this->t('This is the custom page content.'),
+    ];
+
+    return $build;
+  }
+
+  public function checkLoggedIn() {
```

Comment:
If our requirement here is to redirect user after login then this shouldn't be the approach. There are some event available for ex : $events['user.login'] as well user few hooks available for ex hook_user_login . So please use that as per your requirement.